Develop

DorianSyruss opened the pull request.

Commit:
Add initial config files, basic server and routing

- linters, gitignore, editorconfig, express server, routes

Commit:
Add basic schemas, models, routes

- added nodemon for auto server restart
- added 2 basic routs quotes, users
- get i post methods for quotes

Commit:
Add more API end points, global error handling

- added more api routes
- errs propagating to next middleware to global err handler

Commit:
Improve project structure, implement router

- code restructred to functional units, added api and models folders
- add more routues and api end points
- router implemented

Commit:
Add QuoteCollection model, env config, API/index.js

- app.js refactored

Commit:
Add more  model routes, add more API points, WIP

- added more user routes and api points,
- add option to bookmark quotes and
create quote collections,
- poslih app.js,
- code refactored

Commit:
Add option to list all quotes inside of collection

- promise.all, generators, yield implemented

Commit:
Polish routes, WIP

- new routes added, old routes polished

Commit:
env setup, routes refactor, models refactor WIP

- added more api routes,
- refactor api routes code,
- setup production env with remote db on mongolab

Commit:
facebook login, WIP

Commit:
facebook login/auth WIP, WIP

- code is written in poor manners, for testing purposes
- login passes, but no user is written in db, needs testing and
refactoring

Commit:
Add facebook login, code refactored

Commit:
Remove .env.prod from git

Commit:
Delete .env.prod

vladimyr commented on .editorconfig:
```diff
@@ -0,0 +1,17 @@
+max_line_length = 120
+
+[*.{js,json}]
+max_line_length = off
```
That neat little red marker means: no blank new line at the end of file aka **wrong**

vladimyr commented on .eslintrc:
```diff
@@ -0,0 +1,36 @@
+          "arraysInObjects": false,
+          "objectsInObjects": false,
+          "propertyName": true
+        }],*/
```
Remove commented out stuff and while you are doing that rename whole file to `.eslintrc.json`...

vladimyr reviewed (commented):
@kronicker Now it is your turn. I'm [drain bramaged](http://gigaplus.makeshop.jp/supplycrew/img/brand/braindead.jpg) after IMDB scraper...

kronicker commented on api/collections.js:
```diff
@@ -0,0 +1,95 @@
+
+const QuoteCollection = require('../models/quoteCollection.js');
+const router = require('express').Router();
+const HTTPStatus = require('http-status');
```
Maybe take a convention and stick to it. e.g. npm modules first, then blank space then your modules. also alphabetizing is a good for readability. Naming-wise you can capitalize modules and use lower case for importing single functions

DorianSyruss replied:
Models are named with capital first letter, api end points are not, QuoteCollection (model) vs quoteCollections (api end point), as for the order of including I agree

kronicker commented on api/index.js:
```diff
@@ -0,0 +1,11 @@
+const quotes = require('./quotes.js');
+const users = require('./users.js');
+const quoteCollections = require('./collections.js');
+
```
Look at comment about modules and naming. You are not consistent.

kronicker replied:
Also, no need to use `.js` and be consistent in that

kronicker commented on api/collections.js:
```diff
@@ -0,0 +1,95 @@
+
+const QuoteCollection = require('../models/quoteCollection.js');
+const router = require('express').Router();
+const HTTPStatus = require('http-status');
```
Pick a naming/import convention and stick with it throughout code base. 
My favourite is all npm/third-party modules first then your own modules separated with blank space. Naming-wise, you can capitalize moudule imports and use lower case when importing single functions. Also, alphabetizing each section improves readability.

kronicker commented on api/quotes.js:
```diff
@@ -0,0 +1,45 @@
+
+const Quote = require('../models/quote');
+const router = require('express').Router();
+const HTTPStatus = require('http-status');
```
As before

kronicker commented on api/quotes.js:
```diff
@@ -0,0 +1,45 @@
+router.get('/quotes/:id', getQuote);
+router.post('/quotes', createQuote);
+router.delete('/quotes/:id', deleteQuote);
+router.put('/quotes/:id', updateQuote);
```
In previous routers you ordered routes per endopoints then methods so do it here as well.

kronicker commented on api/users.js:
```diff
@@ -0,0 +1,45 @@
+router.get('/users/:id', getUser);
+router.post('/users', createUser);
+router.delete('/users/:id', deleteUser);
+router.put('/users/:id', updateUser);
```
above comment for both routes and imports

kronicker commented on api/collections.js:
```diff
@@ -0,0 +1,95 @@
+    title: req.body.title,
+    description: req.body.description,
+    category: req.body.category
+  };
```
`let update` is redundant here if you ask me. You can just send req.body to db query. Also that gives you ability to expand this route without code changes (e.g. if you want also to update something else later).
Later, when you add route validation you can make sure you are not receiving incorrect data or extra garbage.
@vladimyr second opinion? If he agrees look for similar occurrences throughout code base.

kronicker commented on api/collections.js:
```diff
@@ -0,0 +1,95 @@
+    .catch(err => next(err));
+}
+
+
```
extra blank space

kronicker commented on app.js:
```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+//require needed dependencies
```
Why this comment only here? If you'd like to separate your files in sections with such comments do it everywhere.

kronicker commented on app.js:
```diff
@@ -0,0 +1,50 @@
+const bodyParser = require('body-parser');
+const HTTPStatus = require('http-status');
+const app = express();
+mongoose.Promise = require('bluebird');
```
All thats been said about imports

kronicker commented on package.json:
```diff
@@ -0,0 +1,30 @@
+  "description": "Wall of Sayings",
+  "main": "index.js",
+  "scripts": {
+    "start": "nodemon app.js",
```
Maybe rename this to `start-dev` or `start:dev` or `start:nodemon` since you won't use nodemon in production

DorianSyruss replied:
👍

kronicker commented on app.js:
```diff
@@ -0,0 +1,50 @@
+
+app.get('/auth/facebook', passport.authenticate('facebook', { scope: "email" }));
+app.get('/auth/facebook/callback', passport.authenticate('facebook',
+  { successRedirect: '/api/users', failureRedirect: '/login' }));
```
You separated all routes to separate files, why not this as well?

kronicker reviewed (commented):
@vladimyr I did my best in short time, fell free to overhead me where needed 👍

kronicker commented on passport/facebook.js:
```diff
@@ -0,0 +1,44 @@
+const credentials = require('../credentials.json');
```
Not sure if calling dir `passport` makes sense since it is just name of the tool you're using, not any kind of functionality. I believe `auth` would be more appropriate.

DorianSyruss replied:
👍

Commit:
PR input changes, facebook auth refactored

- code polished
- facebook auth updated, refactored

Commit:
facebook login refactor

Commit:
Implement local strategy login, WIP

- password needs hashing and comparing

Commit:
Login refactor, add password hashing

- WIP, needs polish

Commit:
Implement cookies

-needs testing

Commit:
Implement login, authorization

- WIP

Commit:
Implement auth routes, auth check, basic user roles

- added roles (user, admin), authentication check
- code structure refactored, code polished
- env.template file implemented, fb data now in env, removed credentials
- added db.js fil
- started on seed.js mock-up data

- not all routes are protected (maybe some kind of whitelist instead of middleweare
function in each route:)

Commit:
Protect private routes, seed data for db (WIP)

Commit:
Add seed data WIP

Commit:
Refactor seed.js

Commit:
Implement frontend login logic and UI, WIP

- bootstrap/vue login form

Commit:
Switch to webpack, refactor to .vue files

- refactor vue components
- added webpack config
- added bootswatch themes
- folder structure

Commit:
Add more UI components, fix reg/login, refactor code, WIP

Commit:
Add navbar (UI), fb signup

- working on UI flow of the page, general mock-up

Commit:
Add validation for auth forms

- code polish

Commit:
Add more user interaction msgs, log/reg validations

- some UI polish and changes

Commit:
Add eslint for .vue, add css constants, more U/I info

- code polish
- added "wrong pass or email" msg
- js linter for vue files

Commit:
Code polish

Commit:
Add Users tab, add user mockup data

Commit:
Add quote categories mock-up, UI polish

Commit:
Add more UI elements, refactor html structure

- add Quote voting, latest quotes
- html code refactored

Commit:
Delete .DS_Store

Commit:
Add quotes, UI changes, WIP

- working on more UI elemets
- working on UI flow

Commit:
Merge branch 'develop' of github.com:DorianSyruss/wall-of-sayings into develop

Commit:
Add more vue componets, UI changes, UI flow logic

- added Popular.vue
- added My quote collections
- added more UI elements, code refactor, UI polish

Commit:
Display my collections according user login

- propagate user through components

Commit:
Add user Logout, WIP

- TODO: notice backend of logout
